src/firebase-resource-manager.ts

```diff
@@ -0,0 +1,156 @@
+      });
+      const projectInfo = await this.pollCloudProjectCreationOperation(
+        response.body.name /* LRO resource name */
+      );
+      spinner.succeed();
+      return projectInfo;
+    } catch (err) {
+      spinner.fail();
+      logger.debug(err.message);
+      throw new FirebaseError(
+        "Failed to create Google Cloud project. See firebase-debug.log for more info.",
+        { exit: 2, original: err }
+      );
+    }
+  }
+
+  /**
+   * @return {Promise} this function returns a promise that resolves to the new cloud project
+   *     information
+   */
+  private async pollCloudProjectCreationOperation(operationResourceName: string): Promise<any> {
+    const pollerOptions: OperationPollerOptions = {
+      pollerName: "Project Creation Poller",
+      apiOrigin: api.resourceManagerOrigin,
+      apiVersion: "v1",
+      operationResourceName,
+    };
+    return await pollOperation<any>(pollerOptions);
+  }
+
+  /**
+   * Send an API request to add Firebase to the Google Cloud Platform project and poll the LRO
+   * to get the new Firebase project information.
+   * @return {Promise} this function returns a promise that resolves to the new firebase project
+   *    information
+   */
+  private async addFirebaseToCloudProject(projectId: string): Promise<any> {
+    const spinner = new OraWrapper("Adding Firebase to Google Cloud project");
+    spinner.start();
+
+    // TODO(caot): Removed when "Deferred Analytics" and "Deferred Location" are launched
+    const timeZone = "America/Los_Angeles";
+    const regionCode = "US";
+    const locationId = "us-central";
+
+    try {
+      const response = await api.request("POST", `/v1beta1/projects/${projectId}:addFirebase`, {
+        auth: true,
+        origin: api.firebaseApiOrigin,
+        timeout: 15 * ONE_SECOND_MILLIS,
+        data: { timeZone, regionCode, locationId },
+      });
+      const projectInfo = await this.pollAddFirebaseToCloudProjectOperation(
+        response.body.name /* LRO resource name */
+      );
+      spinner.succeed();
+      return projectInfo; /* Operation resource name */
+    } catch (err) {
+      spinner.fail();
+      logger.debug(err.message);
+      throw new FirebaseError(
+        "Failed to add Firebase to Google Cloud Platform project. See firebase-debug.log for more info.",
+        { exit: 2, original: err }
+      );
+    }
+  }
+
+  /**
+   * @return {Promise} this function returns a promise that resolves to the new firebase project
+   *    information
+   */
+  private async pollAddFirebaseToCloudProjectOperation(
+    operationResourceName: string
+  ): Promise<any> {
+    const pollerOptions: OperationPollerOptions = {
+      pollerName: "Add Firebase Poller",
+      apiOrigin: api.firebaseApiOrigin,
+      apiVersion: "v1beta1",
+      operationResourceName,
+    };
+    return await pollOperation<any>(pollerOptions);
+  }
+}
+
+export function createFirebaseProject(
```

Comment:
If the plan is to separately export all the public methods in this file, then there might not be a benefit to having a `FirebaseResourceManager` class at all anymore. If you remove the class wrapper, you can just export the functions directly rather than writing out their method signatures twice like this.

Reply:
Done